receiver/k8sobjectsreceiver/unstructured_to_logdata.go

```diff
@@ -67,6 +69,7 @@ func unstructuredListToLogData(event *unstructured.UnstructuredList, config *K8s
 			namespaceResourceMap[e.GetNamespace()] = logSlice
 		}
 		record := logSlice.AppendEmpty()
+		record.SetObservedTimestamp(pcommon.NewTimestampFromTime(time.Now()))
```

Comment:
I wonder if we should be providing time of the observation instead of calling `Now()` on each record? What I mean we could pass the observation time to `pullObjectsToLogData` (`watchObjectsToLogData`) from the caller, this way we ensure that all records actually observed during a single pull (watch) have the same observed timestamp.

Reply:
Yup, that's a good idea.
I'll take a look at how to propagate that information from from the caller.

Reply:
Done 😄